Clustering/Core/src/test/java/org/tribuo/neighbours/TestNeighboursPerf.java

```diff
@@ -0,0 +1,158 @@
+        NeighboursBruteForce nbf = nbfFactory.createNeighboursQuery(data);
+        executeQuery(nbf, 5);
+
+        logger.info("Naive implementation: small dataset, small k");
+        NeighboursBruteForceNaive nbfn = nbfnFactory.createNeighboursQuery(data);
+        executeQuery(nbfn, 5);
+        logger.info("");
+
+        logger.info("Target implementation: small dataset, medium k");
+        executeQuery(nbf, 50);
+
+        logger.info("Naive implementation: small dataset, medium k");
+        executeQuery(nbfn, 50);
+        logger.info("");
+
+        logger.info("Target implementation: small dataset, large k");
+        executeQuery(nbf, 200);
+
+        logger.info("Naive implementation: small dataset, large k");
+        executeQuery(nbfn, 200);
+        logger.info("");
+
+        logger.info("");
+        ////
+
+        data = getVectors(100000, 2L);
+
+        logger.info("Target implementation: medium dataset, small k");
+        nbf = nbfFactory.createNeighboursQuery(data);
+        executeQuery(nbf, 5);
+
+        logger.info("Naive implementation:: medium dataset, small k");
+        nbfn = nbfnFactory.createNeighboursQuery(data);
+        executeQuery(nbfn, 5);
+        logger.info("");
+
+        logger.info("Target implementation: medium dataset, medium k");
+        executeQuery(nbf, 50);
+
+        logger.info("Naive implementation: medium dataset, medium k");
+        executeQuery(nbfn, 50);
+        logger.info("");
+
+        logger.info("Target implementation: medium dataset, large k");
+        executeQuery(nbf, 200);
+
+        logger.info("Naive implementation: medium dataset, large k");
+        executeQuery(nbfn, 200);
+
+        logger.info("");
+        ////
+
+        data = getVectors(200000, 3L);
+
+        logger.info("Target implementation: big dataset, small k");
+        nbf = nbfFactory.createNeighboursQuery(data);
+        executeQuery(nbf, 5);
+
+        logger.info("Naive implementation: big dataset, small k");
+        nbfn = nbfnFactory.createNeighboursQuery(data);
+        executeQuery(nbfn, 5);
+
+        logger.info("Target implementation: big dataset, medium k");
+        executeQuery(nbf, 50);
+
+        logger.info("Naive implementation: big dataset, medium k");
+        executeQuery(nbfn, 50);
+
+        logger.info("Target implementation: big dataset, large k");
+        executeQuery(nbf, 200);
+
+        logger.info("Naive implementation: big dataset, large k");
+        executeQuery(nbfn, 200);
+
+        logger.info("");
+        ////
+    }
+
+    //@Test
+    public void testSingleThreadedQueries() {
+        NeighboursBruteForceFactory nbfFactory = new NeighboursBruteForceFactory(DistanceType.L2, 1);
+        NeighboursBruteForceNaiveFactory nbfnFactory = new NeighboursBruteForceNaiveFactory(DistanceType.L2, 1);
+
+        logger.info("PERFORMING SINGLE THREADED TESTS...");
+        logger.info("");
+        doTestIteration(nbfFactory, nbfnFactory);
+    }
+
+    //@Test
```

Comment:
The test annotations are commented here to avoid slowing down the build by default.